src/Foundatio/Jobs/IQueueJob.cs

```diff
@@ -27,9 +31,29 @@ public static class QueueJobExtensions
 
         using var linkedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
         linkedCancellationTokenSource.CancelAfter(acquireTimeout);
+        bool hasAcquireTimeout = true;
+
+        var logger = job.GetLogger();
 
         // NOTE: This has to be awaited otherwise the linkedCancellationTokenSource cancel timer will not fire.
-        await job.RunUntilEmptyAsync(linkedCancellationTokenSource.Token);
+        await job.RunContinuousAsync(cancellationToken: linkedCancellationTokenSource.Token, continuationCallback: async () =>
```

Comment:
Why is this better? It seems like it's just repeating code.

Reply:
It's slightly better because the current approach cancels RunUntilEmpty after x time so it's not really an acquire timeout but a max wait. This is more of an acquire as it will reset the cancel after the first item is processed, so it doesn't wait. The downside is the first item has to process within the acquire timeout.

Reply:
I don't like that. It's called RunUntilEmpty. The only thing it should care about is when the queue is empty. And the timeout is just how long you are willing to wait for the queue to be empty.

Reply:
So if it runs and gets cancelled should it throw if items are remaining or being processed?

Reply:
Would be nice if it was able to return a count of items that were processed. I don't think it should throw.

Reply:
Yeah, I know... I don't love it either. It really should really only be a max wait overall or max wait for acquiring first item... But the acquiring... not sure that is possible due to how it's written.

Reply:
It should just do what it says. Wait X amount of time for the queue to be empty.